server/src/services/auth.service.ts

```diff
@@ -0,0 +1,51 @@
+import type { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import { Admin } from '../entities/Admin';
+import { AppDataSource } from '../../orm.config';
+import { jwtOptions } from '../config/authConfig';
+import jwt from 'jsonwebtoken';
+
+const generateAccessToken = (username: string): string => {
+  const payload = {
+    username,
+  };
+
+  return jwt.sign(payload, jwtOptions.secret, { expiresIn: '24h' });
+};
+
+export class AuthService {
+  async adminLogin(req: Request, res: Response): Promise<Response> {
+    try {
+      const errors = validationResult(req);
+
+      if (!errors.isEmpty()) {
+        const errorMessage = errors.array()[0].msg;
+
+        return res.status(400).json({ message: errorMessage });
+      }
+
+      const { username, password } = req.body;
+
+      const admin = await AppDataSource.manager.findOneBy(Admin, {
+        username,
+      });
+
+      if (admin === null || admin === undefined) {
+        return res.status(404).json({ message: `User ${username} not found` });
+      }
+
+      if (password !== admin.password) {
+        return res.status(400).json({ message: 'Incorrect password' });
```

Comment:
I think 401 should be here for all errors, as login attempt failed

Reply:
it's best practice to respond with 401 Unauthorized when the provided username is not found during a login request. This indicates that the request is not authorized due to incorrect or missing credentials, without disclosing information about the existence of specific usernames in your system.

Reply:
For clarity purpose, we can keep unique messages but change to 401 as it's more compliant to standard

Reply:
please change 400 and 404 here to 401